### PR TITLE
Add useEmulator() API

### PR DESCRIPTION
**Pull request by shepeliev**
- Added ```useEmulator()``` API for ```FirebaseAuth```, ```FirebaseFirestore```, ```FirebaseDatabase```, ```FurebaseFunctions```. 
- Refactored integration tests for using Firebase emulators.
- ```FirebaseFunction.useFunctionsEmulator()``` deprecated.

### CODE REVIEW

File: firestore.rules [hunk 1]
```diff
@@ -0,0 +1,8 @@
+rules_version = '2';
+service cloud.firestore {
+  match /databases/{database}/documents {
+    match /{document=**} {
+      allow read, write: if true;
+    }
+  }
+}
```
Comment (nbransby) [inline firestore.rules:8]: can this fille be moved to the test dir?

File: firebase.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  },
+  "emulators": {
+    "auth": {
+      "port": 9099
+    },
+    "firestore": {
+      "port": 8080
+    }
+  }
+}
```
Comment (nbransby) [inline firebase.json:13]: can this fille be moved to the test dir?

File: firebase-common/src/androidMain/res/xml/network_security_config.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<network-security-config>
+    <domain-config cleartextTrafficPermitted="true">
+        <domain includeSubdomains="true">10.0.2.2</domain>
+    </domain-config>
+</network-security-config>
```
Comment (nbransby) [inline firebase-common/src/androidMain/res/xml/network_security_config.xml:6]: can this config be added to test target only?

File: firebase-common/src/androidMain/AndroidManifest.xml [hunk 4]
```diff
@@ -1 +1,4 @@
-<manifest package="dev.gitlive.firebase.common"/>
+<manifest package="dev.gitlive.firebase.common"
+    xmlns:android="http://schemas.android.com/apk/res/android">
+    <application android:networkSecurityConfig="@xml/network_security_config"/>
+</manifest>
```
Comment (nbransby) [inline firebase-common/src/androidMain/AndroidManifest.xml:4]: can this config be added to test target only?

File: firebase-common/build.gradle.kts [hunk 5]
```diff
@@ -20,6 +20,7 @@ android {
     sourceSets {
         getByName("main") {
             manifest.srcFile("src/androidMain/AndroidManifest.xml")
+            res.srcDir("src/androidMain/res")
```
Comment (nbransby) [inline firebase-common/build.gradle.kts:23]: can this config be added to test target only?

**Review by nbransby — COMMENTED** [review]
@shepeliev this is great thanks - added a few questions above

File: firestore.rules (hunk 1 above)
Comment (shepeliev) [reply firestore.rules]: Firebase configs are moved to test dir

File: firebase-common/src/androidMain/AndroidManifest.xml (hunk 4 above)
Comment (shepeliev) [reply firebase-common/src/androidMain/AndroidManifest.xml]: Yes, I've added AndroidManifest for test target only.

File: firebase-common/build.gradle.kts (hunk 5 above)
Comment (shepeliev) [reply firebase-common/build.gradle.kts]: This resource is not required anymore. I've removed it.

**Review by nbransby — APPROVED** [review]

**Pull request merged**